classwork/сlasswork2.1/main.cpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <string>
+#include <print>
+
+class UserStatistics {
+public:
+    UserStatistics() {
+        loadFromFile();
```

Comment:
Very good implementation for the first attempt.
But let's improve it a bit.
1. It is reasonable to have something like `explicit UserStatistics(const strd::string& statisticsFilePath)` as a constructor.
Why it is important? We can pass any file we want to this class which is more flexible. As a result we can cover this class with unit tests that will not affect production database file.
Note: `explicit` keyword for constructor allows to avoid implicit conversions from string object e.g.
```
auto someString = std::string("some string");
UserStatistics stats = someString;
```
That keyword is reasonable to use when we have a constructor with a single argument.
2. Let's move "user_stats.txt", "bread", "delete" to some anonymous namespace before this class and we can define them as constexpr named constants instead of hardcoding across the code.
Why it is important? More flexible code, easy to change commands if needed.
3. Another thing to improve: it is better to have entities with a single responsibility. In our case UserStatistics seems to be a class that manages data. But in your implementation we can find a lot of business logic mostly related to message printing. My proposal is to cleanup the class to have some public API related to data manipulation like `bool resetUser`,` int incrementCounterFor` and so on. As a result it will be a responsibility of the code that uses the class to handle results and implement custom business logic. As a result we will be able to easily reuse this class for other programs with no changes except maybe API extension if needed. Also it will be easier to cover such a class with unit tests.